### PR TITLE
tests/xtimer_drift: Compute drift and jitter

### PR DESCRIPTION
**Pull request by gebart**
- drift is the total difference from the expected time
 - jitter is the difference only since the last printout

### CODE REVIEW

File: tests/xtimer_drift/main.c [hunk 1]
```diff
@@ -108,9 +112,12 @@ void *worker_thread(void *arg)
         hr = sec / 3600;
         if ((loop_counter % TEST_HZ) == 0) {
             uint32_t expected = start + loop_counter * TEST_INTERVAL;
-            int32_t diff = now - expected;
-            printf("now=%" PRIu32 ".%06" PRIu32 " (%u hours %u min), diff=%" PRId32 "\n",
-                sec, us, hr, min, diff);
+            int32_t drift = now - expected;
+            expected = last + TEST_HZ * TEST_INTERVAL;
```
Comment (A-Paul) [inline tests/xtimer_drift/main.c]: Is `TEST_HZ * TEST_INTERVAL` calculated once or every loop cycle?

File: tests/xtimer_drift/main.c [hunk 2]
```diff
@@ -89,6 +89,7 @@ void *worker_thread(void *arg)
     (void) arg;
     unsigned int loop_counter = 0;
     uint32_t start = 0;
+    uint32_t last = 0;
```
Comment (A-Paul) [inline tests/xtimer_drift/main.c]: Do you know the reason for mixing native and `stdint.h`types?

File: tests/xtimer_drift/main.c (hunk 1 above)
Comment (gebart) [inline tests/xtimer_drift/main.c]: No, the compiler will calculate the product `TEST_HZ * TEST_INTERVAL` in advance and just write that value in the machine code.

**Comment by A-Paul** [comment]
@gebart, though it is not in the scope of your PR, do you mind to do a `s/1000000/USEC_PER_SEC/g`?

File: tests/xtimer_drift/main.c (hunk 2 above)
Comment (gebart) [inline tests/xtimer_drift/main.c]: `xtimer_now` returns `uint32_t` while the loop counter doesn't really need to be any particular size, I guess I could change it if you want to.

File: tests/xtimer_drift/main.c [hunk 3]
```diff
@@ -108,9 +112,12 @@ void *worker_thread(void *arg)
         hr = sec / 3600;
         if ((loop_counter % TEST_HZ) == 0) {
             uint32_t expected = start + loop_counter * TEST_INTERVAL;
-            int32_t diff = now - expected;
-            printf("now=%" PRIu32 ".%06" PRIu32 " (%u hours %u min), diff=%" PRId32 "\n",
-                sec, us, hr, min, diff);
+            int32_t drift = now - expected;
+            expected = last + TEST_HZ * TEST_INTERVAL;
+            int32_t jitter = now - expected;
+            printf("now=%" PRIu32 ".%06" PRIu32 " (%u hours %u min), drift=%" PRId32 ", jitter=%" PRId32 "\n",
```
Comment (A-Paul) [inline tests/xtimer_drift/main.c]: Maybe provide units for *drift* and *jitter* too?

**Comment by A-Paul** [comment]
A quick "compile & run" on *native* and *arduino-due* was successful.
ACK if travis is also happy.

File: tests/xtimer_drift/main.c (hunk 2 above)
Comment (A-Paul) [inline tests/xtimer_drift/main.c]: Yes please. Prevents others from taking time to make their mind about the distinction. :-)

**Comment by gebart** [comment]
I will address the comments, don't merge yet.

**Comment by authmillenon** [comment]
@gebart ping?

**Comment by gebart** [comment]
updated with new information text for drift and jitter.

**Comment by gebart** [comment]
ping @A-Paul

**Comment by A-Paul** [comment]
Thanks for addressing comments.
Tests passed and ACK.

**Pull request merged**